### PR TITLE
Externals: Update static FFmpeg to 4.2.1 with new features, Add libx264 encoder (Windows Only!)

### PR DESCRIPTION
**Pull request by altimumdelta**
Statically-linked method of FFmpeg integration (inside Dolphin.exe) - pending decision at #8121
The other one, the dynamic-delayed-load method is currently here #7110

Anyway this method/branch will be used for development in the meantime, a change of method should not have a big impact when development is complete.

### Supported Features:
AVI Container format
Matroska container format
MPEG-TS container format
MXF container format
WEBM container format
rawvideo container format
frame container support
ffmetadata container support
MPEG-4 encoding support
H.264 (AVC) encoding support (libx264)
FFV1 encoding support
FFVhuff encoding support
MagicYuv encoding support
BMP image encoding support
rawvideo encoding support

---
This upgrade currently increases the Dolphin's Windows executable a bit more than 2 Megabytes.

### CODE REVIEW

File: Source/Core/DolphinQt/DolphinQt.vcxproj [hunk 1]
```diff
@@ -40,8 +40,9 @@
   </PropertyGroup>
   <ItemDefinitionGroup>
     <Link>
-      <AdditionalLibraryDirectories>$(ExternalsDir)ffmpeg\lib;$(IntDir)..\discord-rpc\bin;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
-      <AdditionalDependencies>avrt.lib;iphlpapi.lib;winmm.lib;setupapi.lib;opengl32.lib;glu32.lib;rpcrt4.lib;comctl32.lib;avcodec.lib;avformat.lib;avutil.lib;swresample.lib;swscale.lib;Shlwapi.lib;discord-rpc.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalLibraryDirectories>$(ExternalsDir)x264-win64\lib;$(ExternalsDir)ffmpeg-win64\lib;$(IntDir)..\discord-rpc\bin;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
+      <AdditionalDependencies>avrt.lib;iphlpapi.lib;winmm.lib;setupapi.lib;opengl32.lib;glu32.lib;rpcrt4.lib;comctl32.lib;avcodec.lib;avformat.lib;avutil.lib;swresample.lib;swscale.lib;x264.lib;bcrypt.lib;Shlwapi.lib;discord-rpc.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <TreatLinkerWarningAsErrors Condition="'$(Configuration)|$(Platform)'=='Release|x64'">false</TreatLinkerWarningAsErrors>
```
Comment (BhaaLseN) [inline Source/Core/DolphinQt/DolphinQt.vcxproj]: Did you add this one on purpose? (as well as the one down in `UnitTests.vcxproj`)

Comment (altimumdelta) [reply Source/Core/DolphinQt/DolphinQt.vcxproj]: That's a leftover from development/testing that slipped through, thanks for catching it.

**Comment by CookiePLMonster** [comment]
@dolphin-emu-bot rebuild

**Comment by altimumdelta** [comment]
Ah, not really working tho, due to some changes in 4.2.1 or around VS2017/Dolphin, I can't build it anymore simply due to two similar link errors at the very end:
 `Error LNK2019 unresolved external symbol ff_golomb_vlc_len referenced in function get_ue_golomb_31 Dolphin C:\GitDEV\GitRepos\Github\dolphin\Source\Core\DolphinQt\avformat.lib(mxfenc.o)`

Spent a day on it, and because other's have expressed positive opinion for the dynamic loading method at #7110, I'm closing it now.

**Pull request closed without merging**